RecFCCeeCalorimeter/src/components/CaloTowerToolFCCee.cpp

```diff
@@ -215,63 +205,62 @@ uint CaloTowerToolFCCee::buildTowers(std::vector<std::vector<float>>& aTowers, b
   const edm4hep::CalorimeterHitCollection* ecalBarrelCells = m_ecalBarrelCells.get();
   debug() << "Input Ecal barrel cell collection size: " << ecalBarrelCells->size() << endmsg;
   // Loop over a collection of calorimeter cells and build calo towers
-  if (m_ecalBarrelSegmentation != nullptr) {
-    CellsIntoTowers(aTowers, ecalBarrelCells, m_ecalBarrelSegmentation, m_ecalBarrelSegmentationType, fillTowersCells);
+  if (m_ecalBarrelSegmentationOK) {
```

Comment:
if we remove the m_XXXSegmentationOK booleans, here we could check instead if the cell collection name was not an empty string. 
In a later PR, if we refactor the code to pass lists of cell collections and readouts, we won't even need to do this check - we would just loop over the list of cell collection names, retrieve the cells and call CellsIntoTowers for each cell collection